### ISSUE TITLE
Style/TopLevelMethodDefinition False Positive

### ISSUE THREAD

**Issue by pbstriker38** [rubocop/rubocop#10808]
Style/TopLevelMethodDefinition is triggering on all rspec files that have `require 'rails_helper'`

--------

## Expected behavior

Style/TopLevelMethodDefinition does not trigger on `require`'s

## Actual behavior

The following spec triggers an offense

```ruby
# frozen_string_literal: true

require 'rails_helper'

RSpec.describe User, type: :model do
  describe '#first_name' do
    let(:user) { FactoryBot.create(:user, name: 'Homer Jay Simpson') }

    it 'calculates first name' do
      expect(user.first_name).to eq 'Homer'
    end
  end
```

rubocop debug output
```sh
❯ bundle exec rubocop spec/models/user_spec.rb --debug
For my-app: configuration from my-app/.rubocop.yml
configuration from /Users/daniel/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rubocop-performance-1.14.2/config/default.yml
configuration from /Users/daniel/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rubocop-performance-1.14.2/config/default.yml
Default configuration from /Users/daniel/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rubocop-1.31.2/config/default.yml
configuration from /Users/daniel/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rubocop-rails-2.15.2/config/default.yml
configuration from /Users/daniel/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rubocop-rails-2.15.2/config/default.yml
configuration from /Users/daniel/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rubocop-rake-0.6.0/config/default.yml
configuration from /Users/daniel/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rubocop-rake-0.6.0/config/default.yml
configuration from /Users/daniel/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rubocop-rspec-2.12.1/config/default.yml
configuration from /Users/daniel/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rubocop-rspec-2.12.1/config/default.yml
Use parallel by default.
Running parallel inspection
Inspecting 1 file
Scanning my-app/spec/models/user_spec.rb
Loading cache from /Users/daniel/.cache/rubocop_cache/66084c5688d8101d97a12a1a5238dca1e5c4dccf/32109a8f69a5b7598185d47a6846cc1d555bbc1e/d2ed8237fc89f030eb900cd0a8dae3146a54b057
C

Offenses:

spec/models/user_spec.rb:3:1: C: Style/TopLevelMethodDefinition: Do not define methods at the top-level. (https://rubystyle.guide#top-level-methods)
require 'rails_helper'
^^^^^^^^^^^^^^^^^^^^^^

1 file inspected, 1 offense detected
Finished in 0.5681979998480529 seconds
```

## Steps to reproduce the problem

Run rubocop on a rails rspec file

## RuboCop version

```sh
❯ bundle exec rubocop -V
1.31.2 (using Parser 3.1.2.0, rubocop-ast 1.19.1, running on ruby 3.1.2 arm64-darwin21)
  - rubocop-performance 1.14.2
  - rubocop-rails 2.15.2
  - rubocop-rake 0.6.0
  - rubocop-rspec 2.12.1
```

**Comment by pbstriker38** [comment]
I see that this was already reported and fixed in https://github.com/rubocop/rubocop/pull/10796
